src/components/tl/ucp/gather/gather.c

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+#include "config.h"
+#include "gather.h"
+
+// compute the number of occurences of k in n's factorization
+uint32_t calc_valuation(uint32_t n, uint32_t p)
```

Comment:
make it "static inline uint32_t ucc_calc_valuation".
Also lets move it to utils/ucc_math.h - maybe it will be used for other algs